https://github.com/pytorch/pytorch/runs/6728068311?check_suite_focus=true

```
======================================================================
ERROR: test_save_bf16 (__main__.TestAtenXlaTensor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/xla/test/test_operations.py", line 1628, in test_save_bf16
    torch.save(x, tf)
  File "/opt/conda/lib/python3.7/site-packages/torch/serialization.py", line 379, in save
    _save(obj, opened_zipfile, pickle_module, pickle_protocol)
  File "/opt/conda/lib/python3.7/site-packages/torch/serialization.py", line 589, in _save
    pickler.dump(obj)
  File "/opt/conda/lib/python3.7/site-packages/torch/_tensor.py", line 177, in __reduce_ex__
    return self._reduce_ex_internal(proto)
  File "/opt/conda/lib/python3.7/site-packages/torch/_tensor.py", line 223, in _reduce_ex_internal
    return (torch._utils._rebuild_device_tensor_from_numpy, (self.cpu().numpy(),
TypeError: Got unsupported ScalarType BFloat16
```

I believe the test failure is irrelevant. This PR only touches SyncBatchNorm. The failed test created an XLA tensor and tests saving/loading it, which has nothing to do with SyncBatchNorm.

https://github.com/pytorch/xla/blob/2544253e486c6e5cba8a67b5a419087944a627cc/test/test_operations.py#L1624-L1630

--- a/torch/nn/modules/_functions.py
+++ b/torch/nn/modules/_functions.py
@@ -67,11 +67,19 @@ class SyncBatchNorm(Function):
             # world_size * (2C + 1) -> world_size * C, world_size * C, world_size * 1
             mean_all, invstd_all, count_all = torch.split(combined, num_channels, dim=1)
 
-        # remove stats from empty inputs
-        mask = count_all.squeeze(-1) >= 1
-        count_all = count_all[mask]
-        mean_all = mean_all[mask]
-        invstd_all = invstd_all[mask]
+        if not torch.cuda.is_current_stream_capturing():
+            # The lines below force a synchronization between CUDA and CPU, because
+            # the shape of the result count_all depends on the values in mask tensor.
+            # Such synchronizations break CUDA Graph capturing.
+            # See https://github.com/pytorch/pytorch/issues/78549
+            # FIXME: https://github.com/pytorch/pytorch/issues/78656 describes
+            # a better longer-term solution.
+
+            # remove stats from empty inputs
+            mask = count_all.squeeze(-1) >= 1
+            count_all = count_all[mask]
+            mean_all = mean_all[mask]
+            invstd_all = invstd_all[mask]
 
         # calculate global mean & invstd
         mean, invstd = torch.batch_norm_gather_stats_with_counts(

